From 60da3cf55950114d4efc34121430fa9be5ed6d11 Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Sat, 8 Sep 2012 23:25:47 +0530
Subject: [PORT FROM MAIN] Audio stops after seeking AVI clip with MP3 audio (special contents without proper key samples)

BZ: 43872

During seek in AVI, nearest Sync sample was searched both for
audio and video. Sync samples are useful only in case of
video. We can ignore Sync field for non-video tracks.

Change-Id: Id406894f8f382ab00800604564d72c3b20329493
Orig-Change-Id: Id4bed182958c102848583629a299298642b604b4
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Reviewed-on: http://android.intel.com:8080/77934
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index da4fe8e..e29f9ba 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -1240,6 +1240,12 @@ status_t AVIExtractor::getSampleIndexAtTime(
         return OK;
     }
 
+    // Only video track should be considered for Sync samples
+    if (track.mKind != Track::VIDEO) {
+        *sampleIndex = closestSampleIndex;
+        return OK;
+    }
+
     ssize_t prevSyncSampleIndex = closestSampleIndex;
     while (prevSyncSampleIndex >= 0) {
         const SampleInfo &info =
-- 
1.7.4.1

